state.go

```diff
@@ -257,6 +258,37 @@ func (m *Memberlist) probeNode(node *nodeState) {
 		}
 	}
 
+	// Also make an attempt to contact the node directly over TCP. This
+	// helps prevent confused clients who get isolated from UDP traffic
+	// but can still speak TCP (which also means they can possibly report
+	// misinformation to other nodes via anti-entropy), avoiding flapping in
+	// the cluster.
+	fallbackCh := make(chan bool)
+	if node.PMax >= 3 {
+		destAddr := &net.TCPAddr{IP: node.Addr, Port: int(node.Port)}
```

Comment:
I think we can move this all into `sendPingAndWaitForAck` and change the return to a `(bool, error)`. It should be enough to have the bool indicated `didContact`